app/src/main/java/org/mozilla/fenix/perf/StartupManager.kt

```diff
@@ -0,0 +1,280 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.widget.LinearLayout
+import androidx.annotation.VisibleForTesting
+import androidx.navigation.NavDestination
+import androidx.navigation.fragment.NavHostFragment
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.launch
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.feature.privatemode.notification.PrivateNotificationFeature
+import mozilla.components.support.ktx.android.arch.lifecycle.addObservers
+import mozilla.components.support.utils.SafeIntent
+import mozilla.components.support.utils.toSafeIntent
+import mozilla.components.support.webextensions.WebExtensionPopupFeature
+import org.mozilla.fenix.GleanMetrics.Metrics
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.UriOpenedObserver
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.browser.browsingmode.BrowsingModeManager
+import org.mozilla.fenix.browser.browsingmode.DefaultBrowsingModeManager
+import org.mozilla.fenix.components.metrics.BreadcrumbsRecorder
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.customtabs.ExternalAppBrowserActivity
+import org.mozilla.fenix.ext.breadcrumb
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.home.intent.HomeIntentProcessor
+import org.mozilla.fenix.home.intent.StartSearchIntentProcessor
+import org.mozilla.fenix.session.PrivateNotificationService
+import org.mozilla.fenix.theme.DefaultThemeManager
+import org.mozilla.fenix.theme.ThemeManager
+import java.lang.ref.WeakReference
+import kotlin.reflect.KProperty0
+
+class StartupManager {
+
+    @Suppress("TooManyFunctions")
+    companion object {
+        @OptIn(ExperimentalCoroutinesApi::class)
+        @Suppress("LongParameterList", "LongMethod")
+        fun activityOnCreate(
+            activity: HomeActivity,
```

Comment:
I wonder if we can get away with not passing `activity`? If we pass in a context or Activity, this method can basically do anything and it's hard to know what's _really_ happening when it's called. However, if we pass in a subset of components, we can make more well-reasoned guesses about what it's able to change. Maybe it's not worth it though – I think we'd have to get everyone on-board about separating code from Activities (and other android framework methods) and it's hard to do that from a perf team perspective.

Reply:
I would like to avoid that too.  Multiple places in the code I moved uses `HomeActivity` as a variable such as `onActivityCreateEndHome`. I'm not sure how we would manage that one. Also, what do you mean by a "subset of components"?

Reply:
> Also, what do you mean by a "subset of components"?

For example, instead of `components`, pass `components.strictMode` and `components.intentProcessor` (that's not what we need in this case but this is just an example).